src/components/templates/price-list-table/index.tsx

```diff
@@ -0,0 +1,271 @@
+        handleSearch={setQuery}
+        searchValue={query}
+        {...getTableProps()}
+        className={clsx({ ["relative"]: isLoading })}
+      >
+        <Table.Head>
+          {headerGroups?.map((headerGroup, index) => (
+            <Table.HeadRow {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map((col, headerIndex) => (
+                <Table.HeadCell {...col.getHeaderProps()}>
+                  {col.render("Header")}
+                </Table.HeadCell>
+              ))}
+            </Table.HeadRow>
+          ))}
+        </Table.Head>
+
+        <LoadingContainer isLoading={isLoading || isRefetching || !price_lists}>
+          <Table.Body {...getTableBodyProps()}>
+            {rows.map((row, rowIndex) => {
+              prepareRow(row)
+              return (
+                <PriceListRow
+                  color={"inherit"}
+                  linkTo={row.original.id}
+                  {...row.getRowProps()}
+                  className="group"
+                  linkTo={row.original.id}
+                  id={row.original.id}
+                  row={row}
+                >
+                  {row.cells.map((cell, index) => {
+                    return cell.render("Cell", { index })
+                  })}
+                </PriceListRow>
+              )
+            })}
+          </Table.Body>
+        </LoadingContainer>
+      </Table>
+      <TablePagination
+        count={count!}
+        limit={queryObject.limit}
+        offset={queryObject.offset}
+        pageSize={queryObject.offset + rows.length}
+        title="Price Lists"
+        currentPage={pageIndex + 1}
+        pageCount={pageCount}
+        nextPage={handleNext}
+        prevPage={handlePrev}
+        hasNext={canNextPage}
+        hasPrev={canPreviousPage}
+      />
+      <Toaster
+        visible={selectedFlatRows.length}
+        duration={Infinity}
+        position="bottom-center"
+        id="price-list-batch-actions"
+      >
+        <TableToasterContainer>
+          <span className="inter-small-semibold text-grey-40 pr-5">
+            {selectedFlatRows.length} entries selected:{" "}
+          </span>
+          <div className="pr-base">
+            <HotKeyAction
+              label="Unpublish"
+              hotKey="U"
+              icon="U"
+              onAction={() => console.log("clicked")}
+            />
+          </div>
+          <div className="pr-base">
+            <HotKeyAction
+              label="Delete"
+              hotKey="backspace"
+              icon={<BackspaceIcon />}
+              onAction={() => console.log("clicked")}
+            />
+          </div>
+        </TableToasterContainer>
+      </Toaster>
+    </div>
+  )
+}
+
+const LoadingContainer = ({ isLoading, children }) => {
+  return isLoading ? (
+    <div className="w-full pt-2xlarge flex items-center justify-center">
+      <Spinner size={"large"} variant={"secondary"} />
+    </div>
+  ) : (
+    children
+  )
+}
```

Comment:
nit: Think this component is quite nifty. Should we refactor it to be its own organism / molecule?